sklearn/tree/tree.py

```diff
@@ -240,8 +256,17 @@ def fit(self, X, y, sample_weight=None, check_input=True,
                              "number of samples=%d" % (len(y), n_samples))
         if self.min_samples_split <= 0:
             raise ValueError("min_samples_split must be greater than zero.")
+        if not 1 < min_samples_split <= n_samples:
+            raise ValueError("min_samples_split must in [2, n_samples]")
         if self.min_samples_leaf <= 0:
             raise ValueError("min_samples_leaf must be greater than zero.")
+        elif 0.5 <= self.min_samples_leaf < 1:
```

Comment:
I'm afraid, I don't understand. The line below says that maximum value of `min_samples_leaf` is `n_samples`, so as a fraction it can vary from (0, 1] right?

Also you can just check `min_samples_leaf` since it is now an integer and remove this line?